### PR TITLE
vmm: skip unsupported migration scene

### PR DESCRIPTION
**Pull request by lisongqian**
vmm: skip unsupported migration scene

Fixes: #6516

This patch will fix the error in CI: https://github.com/cloud-hypervisor/cloud-hypervisor/actions/runs/9364108614/job/25808474744?pr=6508

Signed-off-by: Songqian Li <sionli@tencent.com>

### CODE REVIEW

**Comment by up2wing** [comment]
Perhaps you forgot to modify the DeviceManager::resume() ?

**Comment by lisongqian** [comment]
> Perhaps you forgot to modify the DeviceManager::resume() ?

DeviceManager::resume() has been modified in #6518 .

**Comment by up2wing** [comment]
> > Perhaps you forgot to modify the DeviceManager::resume() ?
> 
> DeviceManager::resume() has been modified in #6518 .

The two PR addressed one problem, merging them in one is better.

**Comment by lisongqian** [comment]
> The two PR addressed one problem, merging them in one is better.

Merged to this one, the other PR is closed.

File: vmm/src/device_manager.rs [hunk 1]
```diff
@@ -4816,7 +4816,12 @@ impl Pausable for DeviceManager {
     fn pause(&mut self) -> result::Result<(), MigratableError> {
         for (_, device_node) in self.device_tree.lock().unwrap().iter() {
             if let Some(migratable) = &device_node.migratable {
-                migratable.lock().unwrap().pause()?;
+                match migratable.lock().unwrap().pause() {
```
Comment (up2wing) [inline vmm/src/device_manager.rs:4819]: Does this need to be modified? After all, we only want to resolve the issue of Drop for DeviceManager.

File: vmm/src/device_manager.rs [hunk 2]
```diff
@@ -4857,7 +4867,16 @@ impl Snapshottable for DeviceManager {
         for (_, device_node) in self.device_tree.lock().unwrap().iter() {
             if let Some(migratable) = &device_node.migratable {
                 let mut migratable = migratable.lock().unwrap();
-                snapshot.add_snapshot(migratable.id(), migratable.snapshot()?);
+                let snap = match migratable.snapshot() {
```
Comment (up2wing) [inline vmm/src/device_manager.rs:4870]: Ditto.

File: vmm/src/device_manager.rs [hunk 3]
```diff
@@ -4816,7 +4816,12 @@ impl Pausable for DeviceManager {
     fn pause(&mut self) -> result::Result<(), MigratableError> {
         for (_, device_node) in self.device_tree.lock().unwrap().iter() {
             if let Some(migratable) = &device_node.migratable {
-                migratable.lock().unwrap().pause()?;
+                match migratable.lock().unwrap().pause() {
+                    Err(e) if !matches!(e, MigratableError::UnSupported(_)) => {
+                        return Err(e);
+                    }
+                    _ => {}
```
Comment (likebreath) [inline vmm/src/device_manager.rs:4823]: This is wrong. Why are you ignoring the error type you just added?

**Review by likebreath — CHANGES_REQUESTED** [review]
I am not following the purpose of this PR. What's the reason of adding a new error type `MigratableError::UnSupported`? 

Based on the code, it looks like the whole purpose of adding  `MigratableError::UnSupported` is to ignore it, which is totally wrong.

**Comment by likebreath** [comment]
This is not the right fix to the issue observed from #6508, as explained from https://github.com/cloud-hypervisor/cloud-hypervisor/pull/6508#issuecomment-2153101863.

**Pull request closed without merging**